### PR TITLE
Protect resources controlled by

### PR DESCRIPTION
**Pull request by akucharska**
**Description**

To set it up add:
```
                    {
                      match: {
                        "$.metadata.ownerReferences[0].kind": "^[a-zA-Z]([a-zA-Z0-9_-]*)$",
                      },
                      regex: true,
                    },      
```
to the `PROTECTED_RESOURCES` configuration.

Changes proposed in this pull request:

- ...
- ...
- ...

**Related issue(s)**

### CODE REVIEW

File: docs/features.md [hunk 1]
```diff
@@ -222,9 +224,12 @@ The document lists and describes all the feature flags that are available in Kym
           "messageSrc": "$.metadata.annotations.protected-message"
         },
         {
+          # matches a resource with label "foo" equal to alphanumeric value
           "match": {
-            "$.metadata.labels.foo": "bar"
-          }
+            "$.metadata.labels.foo": "^[a-zA-Z0-9]$"
+          },
+          # allows comparition by regex
```
Comment (Wawrzyn321) [inline docs/features.md:231]:
```suggestion
          # allows comparison by regex
```

File: tests/tests/test-protected-resources.spec.js [hunk 2]
```diff
@@ -90,6 +102,48 @@ context('Test Protected Resources', () => {
...
+
+    cy.getIframeBody()
+      .contains('tr', NAME)
+      .find('[aria-label="Delete"]')
+      .should('be.disabled')
+      .click({ force: true });
+
+    cy.getIframeBody()
+      .contains(`Delete ${NAME}`)
+      .should('not.exist');
```
Comment (Wawrzyn321) [inline tests/tests/test-protected-resources.spec.js:144]: Wouldn't it be sufficient to just check if that button is disabled?

**Review by Wawrzyn321 — COMMENTED** [review]
Apart from those little comments, looks good. I wonder if we should add a tooltip message (instead of generic one) - "This Resource is controlled by other Resource"? It's not a requirement, of course.

**Review by Wawrzyn321 — DISMISSED** [review]

File: docs/features.md [hunk 3]
```diff
@@ -195,6 +195,8 @@ The document lists and describes all the feature flags that are available in Kym
 - **PROTECTED_RESOURCES** – is used to block the edit and delete functions based on the determined rules. If the resource meets the rule requirements, the resource becomes protected and cannot be edited/deleted.  
   Each resource requires the **match** field, which includes a list of key-value pairs. The proper rule description is when the definition given in the key matches the value.
 
+  You can set the **regex** parameter to `true`, to switch comparison mode from **standard** to **regex**.
```
Comment (grego952) [inline docs/features.md:198]:
```suggestion
  You can set the **regex** parameter to `true` to switch comparison mode from **standard** to **regex**.
```

File: docs/features.md [hunk 4]
```diff
@@ -195,6 +195,8 @@ The document lists and describes all the feature flags that are available in Kym
 - **PROTECTED_RESOURCES** – is used to block the edit and delete functions based on the determined rules. If the resource meets the rule requirements, the resource becomes protected and cannot be edited/deleted.  
   Each resource requires the **match** field, which includes a list of key-value pairs. The proper rule description is when the definition given in the key matches the value.
 
+  You can set the **regex** parameter to `true` in order to switch comparison mode from **standard** to **regex**.
```
Comment (NHingerl) [inline docs/features.md:198]:
```suggestion
  To switch comparison mode from **standard** to **regex**, set the **regex** parameter to `true`.
```

**Review by Wawrzyn321 — APPROVED** [review]

**Review by NHingerl — APPROVED** [review]

**Pull request merged**